specification/authorization/resource-manager/Microsoft.Authorization/preview/2024-02-01-preview/common-types.json

```diff
@@ -0,0 +1,491 @@
+          "x-ms-enum": {
+            "name": "RecipientType",
+            "modelAsString": true
+          }
+        },
+        "notificationRecipients": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of notification recipients."
+        },
+        "isDefaultRecipientsEnabled": {
+          "type": "boolean",
+          "description": "Determines if the notification will be sent to the recipient type specified in the policy rule."
+        }
+      }
+    },
+    "RoleManagementPolicyRuleTarget": {
+      "description": "The role management policy rule target.",
+      "type": "object",
+      "properties": {
+        "caller": {
+          "type": "string",
+          "description": "The caller of the setting."
+        },
+        "operations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The type of operation."
+        },
+        "level": {
+          "type": "string",
+          "description": "The assignment level to which rule is applied."
+        },
+        "targetObjects": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of target objects."
+        },
+        "inheritableSettings": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of inheritable settings."
+        },
+        "enforcedSettings": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of enforced settings."
+        }
+      }
+    },
+    "RoleManagementPolicyPimOnlyModeRule": {
+      "description": "The role management policy PIM only mode rule.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/RoleManagementPolicyRule"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "pimOnlyModeSettings": {
+          "$ref": "#/definitions/PIMOnlyModeSettings",
+          "description": "The PIM Only Mode settings"
+        }
+      }
+    },
+    "PIMOnlyModeSettings": {
+      "description": "The PIM Only Mode settings.",
+      "type": "object",
+      "properties": {
+        "mode" : {
+          "type": "string",
+          "description": "Determines whether the setting is enabled, disabled or report only.",
+          "enum": [
+            "Disabled",
+            "Enabled",
+            "ReportOnly"
+          ],
+          "x-ms-enum": {
+            "name": "PIMOnlyMode",
+            "modelAsString": true
+          }
+        },
+        "excludes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-ms-identifiers": [],
+          "description": "The list of object IDs that the rule does not apply to."
+        },
+        "excludeAllServicePrincipals": {
```

Comment:
>excludeAllServicePrincipals [](http://example.com/codeflow?start=9&length=27)

Recommend enum over boolean. Maybe something like: exclude: { None, ServicePrincipals, ... } or something better you come up with.